services/dqa-secondary-reports/main.py

```diff
@@ -0,0 +1,238 @@
+        interval *= 60
+
+        def poll():
+            self.update()
+            threading.Timer(interval, poll).start()
+
+        return poll
+
+    def get_data(self, version):
+        "Entrypoint for Flask routing for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.field_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return self.wrap_response(info)
+
+        return versioned_data
+
+    def get_dict(self):
+        "Entrypoint for Flask routing for dictionary of DQA status codes"
+        try:
+            info = self.dictionary
+        except HTTPError:
+            return 503, ''
+
+        return self.wrap_response(info)
+
+    def get_site_totals(self, version):
+        "Entrypoint for Flask routing for site totals for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.site_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return self.wrap_response(info)
+
+        return versioned_data
+
+    def get_table_totals(self, version):
+        "Entrypoint for Flask routing for table totals for a given version of PEDSnet data model"
+        def versioned_data():
+            try:
+                info = self.table_totals[version]
+            except HTTPError:
+                return 503, ''
+
+            return self.wrap_response(info)
+
+        return versioned_data
+
+    def wrap_response(self, info):
+        content = json.dumps(info, indent=4, sort_keys=True)
+
+        resp = Response(content)
+
+        resp.headers['Content-Type'] = 'application/json'
+
+        return resp
+
+    def get_versions(self):
+        return list(self.field_totals.keys())
+
+
+if __name__ == '__main__':
+    usage = """PEDSnet DQA Secondary Reports Service
+
+    Usage: main.py [--token=<token>] [--host=<host>] [--port=<port>] [--debug] [--interval=<interval>]
+
+    Options:
+        --help              Display the help.
+        --token=<token>     GitHub authorization token.
+        --host=<host>       Host of the service.
+        --port=<port>       Port of the service [default: 5000].
+        --debug             Enable debug output.
+        --interval=<interval>   Time interval, in minutes, to periodically check the repo for updates [default: 30]
+    """
+
+    from docopt import docopt
+
+    opts = docopt(usage)
+
+    host = opts['--host']
+    port = int(opts['--port'])
+    debug = opts['--debug']
+    interval = int(opts['--interval'])
+
+    # Set token if defined.
+    if opts['--token']:
+        GITHUB_AUTH_TOKEN = opts['--token']
+    else:
+        GITHUB_AUTH_TOKEN = os.environ.get('GITHUB_AUTH_TOKEN')
+
+    if not GITHUB_AUTH_TOKEN:
+        print('Authorization token required.')
+        sys.exit(1)
+
+    if debug:
+        logger.setLevel(logging.DEBUG)
+
+    pedsnetDQA = DqaResource()
```

Comment:
Done, thanks. Sometimes I get confused with javascript and python, which one wants camel case and which one wants underscores :-D